src/dune_lang/value.mli

```diff
@@ -35,6 +35,40 @@ module L : sig
   val paths : Path.t list -> t list
   val deps_only : t list -> Path.t list
   val dirs : Path.t list -> t list
-  val concat : t list -> dir:Path.t -> string
   val to_strings : t list -> dir:Path.t -> string list
 end
+
+module Deferred : sig
+  type value = t
+
+  (** A computation which produces a [value] when forced. Forcing requires a
```

Comment:
The description it's a computation, but it really is just a value. It can be compared, serialized, etc. A computation is a function which thankfully it is not.

Reply:
Reworded this comment